org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/commons/ParentProcessWatcher.java

```diff
@@ -83,14 +83,28 @@ private boolean parentProcessStillRunning() {
 		} else {
 			command = "ps -p " + pid;
 		}
+		Process process = null;
+		boolean result = false;
 		try {
-			Process process = Runtime.getRuntime().exec(command);
-			int processResult = process.waitFor();
-			return processResult == 0;
+			process = Runtime.getRuntime().exec(command);
+			result = process.waitFor(POLL_DELAY_SECS, TimeUnit.SECONDS);
+			if (!result) {
+				process.destroy();
+				result = process.waitFor(POLL_DELAY_SECS, TimeUnit.SECONDS);

```

Comment:
why do you wait twice?

Reply:
On SO, this [answer](https://stackoverflow.com/a/5483976/753170) mentions having to read the streams. Does it apply here?

Reply:
No, it doesn't.

Reply:
> why do you wait twice?

We wait for the process to stop.